File: arrays/deleting_items_from_an_array/remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.cpp
/**
 * Author: Nathan Bockisch
 * Date: September 29, 2021
 **/

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }
        
        int new_len = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != nums[new_len]) {
                new_len++;
                nums[new_len] = nums[i];
            }
        }        
        
        return new_len + 1;
    }
};

File: array_and_string/conclusion/remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.rs
/*
 * Author: Nathan Bockisch
 * Date: July 15, 2022
 */
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut k = 0;

        for cur in 0..nums.len() {
            if nums[cur] != nums[k] {
                k += 1;
                nums[k] = nums[cur];
            }
        }

        (k + 1) as i32
    }
}
